python/cudf/cudf/core/frame.py

```diff
@@ -573,6 +582,18 @@ def equals(self, other, **kwargs):
         else:
             return self._index.equals(other._index)
 
+    def _explode(self, explode_column_num: int, index: Optional[cudf.Index]):
+        if index is not None:
+            explode_column_num += index.nlevels
+        res_tbl = libcudf.lists.explode_outer(
+            cudf._lib.table.Table(self._data, index=index), explode_column_num
+        )
+
+        res = self.__class__._from_table(res_tbl)
+        if index is not None:
+            res.index.names = index.names
```

Comment:
Do we need to similarly handle the case of multi-columns here?

Reply:
Need further clarification, you mean `MultiIndex`?

Reply:
Yes, when `self.columns` is a `MultiIndex` specifically because we had a hierarchy of columns